test/index.js

```diff
@@ -91,63 +91,110 @@ test('fail invalid JSON body', function(t) {
       signaturecertchainurl: 'https://invalid'
     },
     body: 'invalid'
-  });
+  })
 
-  t.equal(mockRes.statusCode, 401);
+  t.equal(mockRes.statusCode, 401)
   t.deepEqual(JSON.parse(mockRes._getData()), {
     reason: 'request body invalid json',
     status: 'failure'
-  });
+  })
 
   t.end()
-});
-
+})
 
 test('fail invalid signature', function(t) {
   var mockRes = invokeMiddleware({
     headers: {
       signature: 'aGVsbG8NCg==',
-      signaturecertchainurl: 'https://s3.amazonaws.com/echo.api/echo-api-cert.pem'
+      signaturecertchainurl: 'https://s3.amazonaws.com/echo.api/echo-api-cert-4.pem'
     },
     body: JSON.stringify({
       request: {
         timestamp: new Date().getTime()
       }
     })
-  });
+  }, function() {
+    calledNext = true
+  })
 
+  var calledNext = false
   setTimeout(function() {
-    t.equal(mockRes.statusCode, 401);
+    t.equal(calledNext, false)
+    t.equal(mockRes.statusCode, 401)
     t.deepEqual(JSON.parse(mockRes._getData()), {
-      reason: 'certificate expiration check failed',
+      reason: 'certificate verification failed',
       status: 'failure'
-    });
-    t.end();
-  }, 2000);
-});
+    })
+    t.end()
+  }, 2000)
+})
 
+test('pass with correct signature', function(t) {
+  var timeout = global.setTimeout // see https://github.com/sinonjs/sinon/issues/269
+  var ts = '2017-02-10T07:27:59Z'
+  var now = new Date(ts)
+  var clock = sinon.useFakeTimers(now.getTime())
+  var calledNext = false
+  var mockRes = invokeMiddleware({
```

Comment:
One tiny thing, it might be better to keep huge JSON objects in a separate file to load from, so that if we have future tests that need the same object, the tests themselves will still be readable.